tests/qemu-tests/tests.sh

```diff
@@ -216,13 +209,19 @@ test_system() {
     pkill qemu-system
 }
 
-for platform in "${PLATFORMS[@]}"; do
-    read -r arch kernel_type qemu_args <<< "$platform"
+for vmlinux in "${VMLINUX_LIST[@]}"; do
 
-    ARCH="$arch"
-    KERNEL_TYPE="$kernel_type"
-    QEMU_ARGS=($qemu_args)
-    VMLINUX="${IMAGE_DIR}/vmlinux-${KERNEL_TYPE}-${ARCH}"
+    VMLINUX="${IMAGE_DIR}/${vmlinux}"
+    KERNEL=$(echo "${vmlinux}" | sed "s/vmlinux-//")
+    # extract architecture as last dash-separated group of the kernels name
+    ARCH="${KERNEL##*-}"
+    QEMU_ARGS=()
 
     test_system
+
+    if [[ "${ARCH}" == @("x86_64") ]]; then
+        # additional test with extra QEMU flags
+        QEMU_ARGS=(-cpu qemu64,+la57)
```

Comment:
I guess in the future we'll want to make this more flexible so we can specify different sets of qemu arguments with different kernel images (i.e. `nokaslr`) without having to add an if-statement for each one, but this is fine for now.